app/actions/views/channel.js

```diff
@@ -351,3 +361,51 @@ export function setChannelDisplayName(displayName) {
         displayName
     };
 }
+
+// Returns true if there are more posts to load
+export function increasePostVisibility(channelId, focusedPostId) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {loadingPosts, postVisibility} = state.views.channel;
+        const currentPostVisibility = postVisibility[channelId] || 0;
+
+        if (loadingPosts[channelId]) {
+            return true;
+        }
+
+        dispatch(batchActions([
+            {
+                type: ViewTypes.LOADING_POSTS,
+                data: true,
+                channelId
+            }
+        ]));
+
+        const page = Math.floor(currentPostVisibility / POST_INCREASE_AMOUNT);
+
+        let posts;
+        if (focusedPostId) {
+            posts = await getPostsBefore(channelId, focusedPostId, page, POST_INCREASE_AMOUNT)(dispatch, getState);
+        } else {
+            posts = await getPosts(channelId, page, POST_INCREASE_AMOUNT)(dispatch, getState);
+        }
+
+        if (posts) {
+            // make sure to increment the posts visibility
+            // only if we got results
+            dispatch({
+                type: ViewTypes.INCREASE_POST_VISIBILITY,
+                data: channelId,
+                amount: POST_INCREASE_AMOUNT
+            });
+        }
+
+        dispatch({
+            type: ViewTypes.LOADING_POSTS,
+            data: false,
+            channelId
+        });
+
+        return posts && posts.order.length >= POST_INCREASE_AMOUNT;
```

Comment:
Does anything bad happen if the `getPosts` call fails for whatever reason and this returns false? Does it prevent you from ever getting those posts again?